### DISCUSSION TITLE
Working around symlinks

### DISCUSSION THREAD

**Discussion by dgudim** [twpayne/chezmoi#3520] [Show and tell]
I wanted to move my `.config` and `.cache` to a faster nvme drive.

The first logical action was to move the directory over to the new drive and symlink it back, so I did.
But chezmoi was not particularly happy with that, the symlinks got treated as symlinks and not directories (duh) and chezmoi prompted me with this:
```
diff --git a/.config b/.config
old mode 100000
new mode 40755
index df5c7ebe0b7ba9ce6b7b05a5c416412dcbfd7bb5..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
--- a/.config
+++ b/.config
@@ -1 +0,0 @@
-/mnt/fast_storage/FAST_CONFIG
\ No newline at end of file
```

Soooo, it wanted to overwrite the symlink with a real directory, I needed a different solution.

Bind mounts to the rescue!

I remembered that on linux there is such thing as a bind mount, which basically allows to mount a directory as if it was a drive! 

So I went and added 2 lines to my fstab:
```
/mnt/fast_storage/FAST_CONFIG/  /home/kloud/.config/          none    defaults,bind   0 0
/mnt/fast_storage/FAST_CACHE/   /home/kloud/.cache/           none    defaults,bind   0 0
```

Now I am happy and chezmoi is happy 😎. 

Maybe this helps someone.

**Comment by twpayne** [comment]
Very neat, thanks for sharing :)

**Comment by halostatue** [comment]
Windows users *may* be able to achieve similar behaviour using junction points, and on macOS using `diskutil mount -mountPoint` may help — but these may require a drive, not a directory. Further exploration by motivated parties would be welcome on this.

**Reply by Lockszmith-GH** [reply]
PowerShell command to create the junction point:
```ps1
New-Item -Type Junction -Value 'Target\Path' .\sym-link-name
```
